### DISCUSSION TITLE
AssertionError: Differential drive requires both wheel joints to have same min velocities!

### DISCUSSION THREAD

**Discussion by Daniel70-liu** [StanfordVL/BEHAVIOR-1K#953] [Q&A] [answered]
**### This is the test code**
Sorry to bother you, I would like to ask why such an error occurs. Can anyone explain it?
```
import numpy as np
import omnigibson as og
from omnigibson.macros import gm
from omnigibson.sensors import VisionSensor
from omnigibson.utils.ui_utils import KeyboardRobotController
from omnigibson.robots import Fetch, Tiago 

gm.USE_GPU_DYNAMICS = False
gm.ENABLE_FLATCACHE = True

def main(random_selection=False, headless=False, short_exec=False):
    """
    Robot grasping mode demo with selection
    Queries the user to select a type of grasping mode
    """
    og.log.info(f"Demo {__file__}\n    " + "*" * 80 + "\n    Description:\n" + main.__doc__ + "*" * 80)

    
    scene_cfg = dict(type='Scene')
    
    table_cfg = dict(
        type="DatasetObject",
        name="table",
        category="breakfast_table",
        model="lcsizg",
        bounding_box=[0.5, 0.5, 0.8],
        fixed_base=True,
        position=[0.7, -0.1, 0.6],
        orientation=[0, 0, 0.707, 0.707],
    )

    chair_cfg = dict(
        type="DatasetObject",
        name="chair",
        category="straight_chair",
        model="amgwaw",
        bounding_box=None,
        fixed_base=False,
        position=[0.45, 0.65, 0.425],
        orientation=[0, 0, -0.9990215, -0.0442276],
    )

    box_cfg = dict(
        type="PrimitiveObject",
        name="box",
        primitive_type="Cube",
        rgba=[1.0, 0, 0, 1.0],
        size=0.05,
        position=[0.53, -0.1, 0.97],
    )
   
    cfg = dict(scene=scene_cfg, objects=[table_cfg, chair_cfg, box_cfg])

    env = og.Environment(configs=cfg)
    
    
    robot = Fetch(  
        name="Fetch",
        obs_modalities=["rgb"],
        action_type="continuous",
        action_normalize=True,
    )


    og.sim.import_object(robot)  
    robot.set_position([0, 0, 0])
    og.sim.play()
    og.sim.step()
    
    robot.reset()
    robot.keep_still()
    
    
    for sensor in robot.sensors.values():
        if isinstance(sensor, VisionSensor):
            sensor.image_height = 720
            sensor.image_width = 720

    og.sim.viewer_camera.set_position_orientation(
        position=np.array([-2.39951,  2.26469,  2.66227]),
        orientation=np.array([-0.23898481,  0.48475231,  0.75464013, -0.37204802]),
    )

    print("Press ctrl+c to quit")

    max_steps = -1 if not short_exec else 100
    step = 0
    while step != max_steps:
        action = np.random.uniform(-1,1,robot.action_dim)
        for _ in range(10):
            env.step(action)  # Execute actions for the robot
            step += 1

    env.close()

if __name__ == "__main__":
    main()
```

Traceback (most recent call last):
  File "C:\Users\rtxa6000\測試flowise_omnigibson\Fetch_create_test.py", line 130, in <module>
    main()
  File "C:\Users\rtxa6000\測試flowise_omnigibson\Fetch_create_test.py", line 95, in main
    og.sim.step()
  File "C:\Users\rtxa6000\OmniGibson\omnigibson\simulator.py", line 783, in step
    self._non_physics_step()
  File "C:\Users\rtxa6000\OmniGibson\omnigibson\simulator.py", line 639, in _non_physics_step
    obj.initialize()
  File "C:\Users\rtxa6000\OmniGibson\omnigibson\prims\prim_base.py", line 74, in initialize
    self._initialize()
  File "C:\Users\rtxa6000\OmniGibson\omnigibson\robots\fetch.py", line 263, in _initialize
    super()._initialize()
  File "C:\Users\rtxa6000\OmniGibson\omnigibson\robots\manipulation_robot.py", line 216, in _initialize
    super()._initialize()
  File "C:\Users\rtxa6000\OmniGibson\omnigibson\robots\robot_base.py", line 186, in _initialize
    super()._initialize()
  File "C:\Users\rtxa6000\OmniGibson\omnigibson\objects\stateful_object.py", line 154, in _initialize
    super()._initialize()
  File "C:\Users\rtxa6000\OmniGibson\omnigibson\objects\controllable_object.py", line 124, in _initialize
    self._load_controllers()
  File "C:\Users\rtxa6000\OmniGibson\omnigibson\objects\controllable_object.py", line 183, in _load_controllers
    controller = create_controller(**cfg)
  File "C:\Users\rtxa6000\OmniGibson\omnigibson\controllers\__init__.py", line 34, in create_controller
    return controller_cls(**kwargs)
  File "C:\Users\rtxa6000\OmniGibson\omnigibson\utils\python_utils.py", line 90, in wrapper
    func(*values.args, **values.kwargs)
  File "C:\Users\rtxa6000\OmniGibson\omnigibson\controllers\dd_controller.py", line 60, in __init__
    min_vels[0] == min_vels[1]
AssertionError: Differential drive requires both wheel joints to have same min velocities!

**Comment by hang-yin** [comment, marked as answer]
Hello! Thanks for reaching out. This is a little hard for me to tell what the actual problem is, but here should be a simple fix: I see that you based this script off of the grasping mode example - do you mind try loading the robot with a robot config instead of importing it after the environment has been initialized? For example:

```
    scene_cfg = dict(type='Scene')

    robot0_cfg = dict(
        type="Fetch",
        obs_modalities=["rgb"],  # we're just doing a grasping demo so we don't need all observation modalities
        action_type="continuous",
        action_normalize=True,
        grasping_mode=grasping_mode,
    )
    
    table_cfg = dict(
        type="DatasetObject",
        name="table",
        category="breakfast_table",
        model="lcsizg",
        bounding_box=[0.5, 0.5, 0.8],
        fixed_base=True,
        position=[0.7, -0.1, 0.6],
        orientation=[0, 0, 0.707, 0.707],
    )

    chair_cfg = dict(
        type="DatasetObject",
        name="chair",
        category="straight_chair",
        model="amgwaw",
        bounding_box=None,
        fixed_base=False,
        position=[0.45, 0.65, 0.425],
        orientation=[0, 0, -0.9990215, -0.0442276],
    )

    box_cfg = dict(
        type="PrimitiveObject",
        name="box",
        primitive_type="Cube",
        rgba=[1.0, 0, 0, 1.0],
        size=0.05,
        position=[0.53, -0.1, 0.97],
    )
   
    cfg = dict(scene=scene_cfg, robots=[robot0_cfg], objects=[table_cfg, chair_cfg, box_cfg])

    env = og.Environment(configs=cfg)

    robot = env.robots[0]
    robot.set_position_orientation(position=[0, 0, 0])
    robot.reset()
    robot.keep_still()
```
Let me know if this works for you!

**Reply by Daniel70-liu** [reply]
This method can be solved, thank you for your help.
